lib/shopify_app/access_scopes/user_strategy.rb

```diff
@@ -0,0 +1,35 @@
+# frozen_string_literal: true
+
+module ShopifyApp
+  module AccessScopes
+    class UserStrategy
+      class << self
+        def update_access_scopes_for_user_id?(user_id)
+          user_access_scopes = user_access_tokens_by_user_id(user_id)
+          return true unless user_access_scopes
+          configuration_access_scopes != user_access_scopes
+        end
+
+        def update_access_scopes_for_shopify_user_id?(shopify_user_id)
+          user_access_scopes = user_access_tokens_by_shopify_user_id(shopify_user_id)
+          return true unless user_access_scopes
+          configuration_access_scopes != user_access_scopes
+        end
+
+        private
```

Comment:
I don't know how much this might complicate things, but it would be sweet to unify the `shop_strategy` and `user_strategy` under a single API: 

```ruby
module ShopifyApp
  module AccessScopes
    class Strategy
      def update_access_scopes?(*args)
      end
    end
  end
end 
```

This would make it easy for developers to template their own strategies. So for example, the `UserStrategy` could look something like this:

```suggestion
module ShopifyApp
  module AccessScopes
    class UserStrategy
      class << self
       def update_access_scopes?(user_id:, shopify_user_id:)
          return update_access_scopes_for_user_id(user_id) if user_id
          return update_access_scopes_for_shopify_user_id(shopify_user_id) if shopify_user_id
          raise CustomError
        end

        private
        
        def update_access_scopes_for_user_id?(user_id)
          user_access_scopes = user_access_tokens_by_user_id(user_id)
          return true unless user_access_scopes
          configuration_access_scopes != user_access_scopes
        end

        def update_access_scopes_for_shopify_user_id?(shopify_user_id)
          user_access_scopes = user_access_tokens_by_shopify_user_id(shopify_user_id)
          return true unless user_access_scopes
          configuration_access_scopes != user_access_scopes
        end
```